libdnf5/conf/vars.cpp

```diff
@@ -433,12 +377,21 @@ void Vars::load(const std::string & installroot, const std::vector<std::string>
 }
 
 void Vars::detect_vars(const std::string & installroot) {
+    const char * arch = nullptr;
+    if (contains("arch")) {
+        arch = get_value("arch").c_str();
+    }
+    init_lib_rpm(arch);
+
```

Comment:
This is very minor comment - you already have an information for test whether arch is set or not therefore it is not required to ask set_lazy to check it again. 
```
    set_lazy(
        "arch", []() -> auto { return std::make_unique<std::string>(detect_arch()); }, Priority::AUTO);
```